classes/queue_process.php

```diff
@@ -86,19 +86,16 @@ public function create_client($handler=null) {
             ];
         }
 
-        // If use proxy is configured, add to args.
-        if ($this->config->useproxy) {
-            $connectionoptions['http'] = ['proxy' => \local_aws\local\aws_helper::get_proxy_string()];
-        }
-
-        // Allow handler overriding for testing.
-        if ($handler != null) {
+        // We should use the test handler if provided.
+        if (!empty($handler)) {
             $connectionoptions['handler'] = $handler;
         }
 
         // Only create client if it hasn't already been done.
         if ($this->client == null) {
```

Comment:
Very much not included in the changes but I think this would be better, that and since this is not doing the `===` check for some reason??

```suggestion
        if (is_null($this->client)) {
```

Alternatives include doing `!isset($this->client)` which could be faster technically since isset is a language construct.